rmw_fastrtps_cpp/src/functions.cpp

```diff
@@ -837,12 +861,31 @@ rmw_publisher_t * rmw_create_publisher(const rmw_node_t * node,
     _register_type(participant, info->type_support_, info->typesupport_identifier_);
   }
 
-  PublisherAttributes publisherParam;
-  publisherParam.topic.topicKind = NO_KEY;
-  publisherParam.topic.topicDataType = type_name;
-  publisherParam.topic.topicName = topic_name;
   publisherParam.qos.m_publishMode.kind = ASYNCHRONOUS_PUBLISH_MODE;
   publisherParam.historyMemoryPolicy = PREALLOCATED_WITH_REALLOC_MEMORY_MODE;
+  publisherParam.topic.topicKind = NO_KEY;
+  publisherParam.topic.topicDataType = type_name;
+  // set topic and partitions
+  name_tokens = rcutils_split_last(topic_name, '/');
+  if (name_tokens.size == 1) {
+    publisherParam.qos.m_partition.push_back(ros_topics_prefix);
+    publisherParam.topic.topicName = name_tokens.data[0];
+  } else if (name_tokens.size == 2) {
+    publisherParam.qos.m_partition.push_back(
+      (std::string(ros_topics_prefix) + "/" + name_tokens.data[0]).c_str());
+    publisherParam.topic.topicName = name_tokens.data[1];
+  } else {
+    RMW_SET_ERROR_MSG("Illformated topic name");
+    goto fail;
+  }
+
+  if (rcutils_string_array_fini(&name_tokens) != RCUTILS_RET_OK) {
+    fprintf(stderr, "Failed to destroy the token string array\n");
+  }
+
+  // fprintf(stderr, "ros topic: %s\n", topic_name);
+  // fprintf(stderr, "partition name: %s\n", publisherParam.qos.m_partition.getNames()[0].c_str());
+  // fprintf(stderr, "dds topic: %s\n", publisherParam.topic.topicName.c_str());
```

Comment:
Please remove these or put a TODO describing what ought to be done with them in the future.